core/Process.js

```diff
@@ -12,11 +14,26 @@ var Process = module.exports = function (name, func) {
   this.fiber = null;
   this.inports = {};
   this.outports = {};
-  this.status = Process.Status.NOT_INITIALIZED;
+  this._status = Process.Status.NOT_INITIALIZED;
   this.ownedIPs = 0;
   this.cbpending = false;
   this.yielded = false;
   this.result = null; // [data, err]
+
+  this.trace('Created with status: ' + Process.Status.__lookup(this._status),this.name);
+  Object.defineProperty(this, 'status', {
+    get: function() { return this._status; },
+    set: function(status) {
```

Comment:
This is all so that we can provide nice, useful tracing around process status changes